### PR TITLE
Refactor/billing lazyload killapiv7

### PR DESCRIPTION
**Pull request by frenautvh**
| Question         | Answer
| ---------------- | ---
| Branch?          | develop
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | MANAGER-6316, MANAGER-6207
| License          | BSD 3-Clause

## Description

billing section apiv7 refactoring + billing section lazy loading

- [x] validates 2api /billing/services

### CODE REVIEW

**Comment by radireddy** [comment]
@frenautvh This will create too many merge conflicts with billing and account migration. https://github.com/ovh/manager/pull/4049

File: packages/manager/apps/dedicated/client/app/billing/autoRenew/actions/update/form/form.service.js [hunk 1]
```diff
@@ -1,21 +1,28 @@
...
+      .get(`${get(service, 'route.url')}/serviceInfos`)
+      .then(({ data }) => get(data, 'possibleRenewPeriod'))
+      .then((possibleRenewPeriod) => {
+        return map(possibleRenewPeriod, (period) => ({
+          period,
+          label: this.$translate.instant(
+            'billing_autorenew_service_update_service_period_value',
+            { month: period },
+          ),
+        }));
```
Comment (marie-j) [inline packages/manager/apps/dedicated/client/app/billing/autoRenew/actions/update/form/form.service.js:24]: No catching ? What if `route.url` is not defined ?

File: packages/manager/apps/dedicated/client/app/billing/autoRenew/autorenew.service.js [hunk 2]
```diff
@@ -38,17 +39,28 @@ export default class {
...
-    return this.OvhApiBillingAutorenewServices.Aapi().query().$promise;
+  checkIfLegacy() {
+    return this.ovhFeatureFlipping
+      .checkFeatureAvailability(['billing:billingServices'])
+      .then((availability) =>
+        this.isLegacy.resolve(
+          !availability.isFeatureAvailable('billing:billingServices'),
+        ),
+      )
+      .catch(() => this.isLegacy.resolve(true));
```
Comment (marie-j) [inline packages/manager/apps/dedicated/client/app/billing/autoRenew/autorenew.service.js:63]: Maybe we could have used a `componentProvider` and use one or the other depending on the type, that way cleaning up would have been easier and no need to necessarly keep all existing code

File: packages/manager/apps/dedicated/client/app/billing/autoRenew/actions/autorenew-blocked/autorenew-blocked.module.js [hunk 3]
```diff
@@ -1,8 +1,9 @@
 import angular from 'angular';
-import 'angular-translate';
+import angularTranslate from 'angular-translate';
```
[the remark this answers is not in the document]
Comment (frenautvh) [reply packages/manager/apps/dedicated/client/app/billing/autoRenew/actions/autorenew-blocked/autorenew-blocked.module.js:2]: i took a look at the main entry and it points to the dist/angular-translate.js which is exporting the string

File: packages/manager/apps/dedicated/client/app/billing/autoRenew/actions/delete/delete.routing.js [hunk 4]
```diff
@@ -59,6 +60,7 @@ export default /* @ngInject */ ($stateProvider) => {
           serviceId: service.domain,
           serviceType: service.serviceType,
           renew: service.renew,
+          route: get(service, 'route.url'),
```
Comment (marie-j) [inline packages/manager/apps/dedicated/client/app/billing/autoRenew/actions/delete/delete.routing.js:63]:
```suggestion
          route: service.route?.url
```
not possible ? (that way we avoid a `_.get`

File: packages/manager/apps/dedicated/client/app/billing/autoRenew/autorenew.service.js [hunk 5]
```diff
@@ -330,11 +331,11 @@ export default class {
...
       : Promise.resolve([]);
-    return agreementsPromise.then(() =>
-      this.updateServices([
-        pick(service, ['serviceId', 'serviceType', 'renew']),
-      ]),
-    );
+    return agreementsPromise.then(() => {
+      const toUpdate = pick(service, ['serviceId', 'serviceType', 'renew']);
+      toUpdate.route = get(service, 'route.url');
+      return this.updateServices([toUpdate]);
```
Comment (cbourgois) [inline packages/manager/apps/dedicated/client/app/billing/autoRenew/autorenew.service.js:329]: IMO we could get rid of lodash: 

```js
const { serviceId, serviceType, renew } = service;
return this.updateServices([{
  serviceId, 
  serviceType, 
  renew, 
  route: service.route?.url,
}]);
````

File: packages/manager/apps/dedicated/client/app/billing/autoRenew/cancel-resiliation/cancel-resiliation.routing.js [hunk 6]
```diff
@@ -21,7 +23,12 @@ export default /* @ngInject */ ($stateProvider) => {
         }
 
         service.cancelResiliation();
-        return BillingAutoRenew.updateService(service);
+        return BillingAutoRenew.updateService({
+          serviceId: service.serviceId,
```
Comment (cbourgois) [inline packages/manager/apps/dedicated/client/app/billing/autoRenew/cancel-resiliation/cancel-resiliation.routing.js:27]: Same here.

File: packages/manager/apps/dedicated/client/app/billing/autoRenew/actions/update/update.controller.js [hunk 7]
```diff
@@ -16,6 +18,18 @@ export default class {
     this.model = {
       agreements: this.autorenewAgreements.length === 0,
     };
+
+    return this.$http
+      .get(`${get(this, 'billingService.route.url')}/serviceInfos`)
```
Comment (cbourgois) [inline packages/manager/apps/dedicated/client/app/billing/autoRenew/actions/update/update.controller.js]: What's happen if no `billingService.route.url` ?

File: packages/manager/apps/dedicated/client/app/billing/autoRenew/actions/update/form/form.service.js (hunk 1 above)
Comment (frenautvh) [reply packages/manager/apps/dedicated/client/app/billing/autoRenew/actions/update/form/form.service.js:24]: just added error catching in the caller (controller)

**Review by antleblanc — APPROVED** [review]

**Comment by frenautvh** [comment]
just rebased on develop & resolved conflicts

**Comment by frenautvh** [comment]
rebase on develop c13bcb18a7e5cd28662ef9cc2158776bde0b6db1

**Comment by frenautvh** [comment]
rebased on develop c5eb003157cc31a07616e0f3ca09a390bebdad84

**Comment by frenautvh** [comment]
rebased on 292c1edf32beec2ba62bc231440882d7f6226b9c

**Comment by frenautvh** [comment]
rebased on d1fa37583337f0133f3fc9bc5d2ebf4312b52043

**Comment by frenautvh** [comment]
rebased on 29f23d772c6ac18080c0f0e5c5c30b3149413a36

**Comment by frenautvh** [comment]
this PR will be split in two parts : lazy load & apiv7 refactoring in order to faciliate QA

**Pull request closed without merging**